torch/CMakeLists.txt

```diff
@@ -347,45 +337,76 @@ install(TARGETS torch
+  target_compile_definitions(test_jit PUBLIC USE_CATCH _FORCE_INLINES)
+  target_include_directories(test_jit PUBLIC
+    "${TORCH_SRC_DIR}/../third_party/catch/single_include"
+    ${ATen_CPU_INCLUDE})
 
-  target_link_libraries(test_jit torch)
+  if (USE_CUDA)
+    target_link_libraries(test_jit ${CUDA_LIBRARIES})
+  endif()
+endif()
 
-  target_include_directories(test_jit PUBLIC
-    "${TORCH_SRC_DIR}/../third_party/catch/single_include")
-
-  # API Tests
-
-  if (NOT NO_API)
-    set(TORCH_API_TEST_DIR "${TORCH_SRC_DIR}/../test/cpp/api")
-
-    add_executable(test_api
-      ${TORCH_API_TEST_DIR}/any.cpp
-      ${TORCH_API_TEST_DIR}/modules.cpp
-      ${TORCH_API_TEST_DIR}/cursor.cpp
-      ${TORCH_API_TEST_DIR}/integration.cpp
-      ${TORCH_API_TEST_DIR}/main.cpp
-      ${TORCH_API_TEST_DIR}/misc.cpp
-      ${TORCH_API_TEST_DIR}/module.cpp
-      ${TORCH_API_TEST_DIR}/optim.cpp
-      ${TORCH_API_TEST_DIR}/sequential.cpp
-      ${TORCH_API_TEST_DIR}/rnn.cpp
-      ${TORCH_API_TEST_DIR}/serialization.cpp
-      ${TORCH_API_TEST_DIR}/static.cpp
-      ${TORCH_API_TEST_DIR}/tensor.cpp
-      ${TORCH_API_TEST_DIR}/tensor_cuda.cpp
-      # Temporary until ATen tests are built with Caffe2
-      ${TORCH_API_TEST_DIR}/tensor_options.cpp
-      ${TORCH_API_TEST_DIR}/tensor_options_cuda.cpp
-    )
+if (NOT NO_API AND NOT USE_ROCM)
+  target_include_directories(torch PUBLIC
+    ${TORCH_SRC_DIR}/csrc/api
+    ${TORCH_SRC_DIR}/csrc/api/include)
 
-      target_include_directories(test_api
-        PUBLIC
-        "${TORCH_SRC_DIR}/../third_party/catch/single_include")
+  if (NOT MSVC)
+    target_compile_options(torch PRIVATE -Wno-error=maybe-uninitialized)
+  endif()
+
+  if (APPLE)
+    target_compile_options(torch PRIVATE -Wno-error=unknown-warning-option)
+  endif()
+endif()
 
-    target_link_libraries(test_api torch)
+if (NOT NO_API AND NOT USE_ROCM)
+  set(TORCH_API_TEST_DIR "${TORCH_SRC_DIR}/../test/cpp/api")
+
+  add_executable(test_api
+    ${TORCH_API_TEST_DIR}/any.cpp
+    ${TORCH_API_TEST_DIR}/modules.cpp
+    ${TORCH_API_TEST_DIR}/cursor.cpp
+    ${TORCH_API_TEST_DIR}/integration.cpp
+    ${TORCH_API_TEST_DIR}/main.cpp
+    ${TORCH_API_TEST_DIR}/misc.cpp
+    ${TORCH_API_TEST_DIR}/module.cpp
+    ${TORCH_API_TEST_DIR}/optim.cpp
+    ${TORCH_API_TEST_DIR}/sequential.cpp
+    ${TORCH_API_TEST_DIR}/rnn.cpp
+    ${TORCH_API_TEST_DIR}/serialization.cpp
+    ${TORCH_API_TEST_DIR}/static.cpp
+    ${TORCH_API_TEST_DIR}/tensor.cpp
+    ${TORCH_API_TEST_DIR}/tensor_cuda.cpp
+    # Temporary until ATen tests are built with Caffe2
+    ${TORCH_API_TEST_DIR}/tensor_options.cpp
+    ${TORCH_API_TEST_DIR}/tensor_options_cuda.cpp
+    )
+
+  target_include_directories(test_api
+    PUBLIC
+    "${TORCH_SRC_DIR}/../third_party/catch/single_include"
+    ${ATen_CPU_INCLUDE})
+
+  target_link_libraries(test_api torch ${TORCH_CUDA_LIBRARIES})
+
+  if (APPLE)
+    target_compile_options(test_api PRIVATE
+      -Wno-error=unknown-warning-option
```

Comment:
I'm actually not a fan of this. I want my build to be as clean as possible. If we're not going to fix the warning, let's disable it and not have it spew in our build (i.e. I'd rather `Wno-missing-braces`). I or someone else will fix the warnings eventually.

That said, `Wno-reorder` is just plain evil. It's just inviting free bugs when you use a member in the initialization of another member.
`Wmaybe-uninitialized` should be fixed with `(void)variable` and we have in the past.

Why are you adding these new warning disablements anyway? The code being compiled didn't change, did it?

Reply:
agreed, however see my response above

Reply:
You don't have to fix the warnings, but do please change them to `-Wno-unknown-warning-option` etc. instead of `-Wno-error=...` so that they don't show up in the build

Reply:
++